src/Views/ProgressView.vala

```diff
@@ -262,58 +256,72 @@ public class ProgressView : AbstractInstallerView {
                     return;
                 }
 
-                start = disk.get_sector (ref efi_sector);
-                end = disk.get_sector (ref end_sector);
+                break;
+        }
 
-                // EFI installs require both an EFI and root partition, so this add a new EXT4
-                // partition that is configured to start at the end of the EFI sector, and
-                // continue to the end of the disk.
-                result = disk.add_partition (
-                    new Distinst.PartitionBuilder (start, end, Distinst.FileSystemType.LVM)
-                        .partition_type (Distinst.PartitionType.PRIMARY)
-                        .logical_volume ("data", encryption)
-                );
+        start = disk.get_sector (ref boot_sector);
+        end = disk.get_sector (ref end_sector);
 
-                if (result != 0) {
-                    warning ("unable to add lvm partition to %s\n", current_config.disk);
-                    on_error ();
-                    return;
-                }
+        // EFI installs require both an EFI and root partition, so this add a new EXT4
+        // partition that is configured to start at the end of the EFI sector, and
+        // continue to the end of the disk.
+        result = disk.add_partition (
+            new Distinst.PartitionBuilder (start, end, Distinst.FileSystemType.LVM)
+                .partition_type (Distinst.PartitionType.PRIMARY)
+                .logical_volume ("data", encryption)
+        );
 
-                disks.push (disk);
+        if (result != 0) {
+            warning ("unable to add lvm partition to %s\n", current_config.disk);
+            on_error ();
+            return;
+        }
 
-                result = disks.initialize_volume_groups ();
+        disks.push (disk);
 
-                if (result != 0) {
-                    warning ("unable to initialize volume groups on %s\n", current_config.disk);
-                    on_error ();
-                    return;
-                }
+        result = disks.initialize_volume_groups ();
 
-                unowned Distinst.LvmDevice lvm_device = disks.find_logical_volume ("data");
+        if (result != 0) {
+            warning ("unable to initialize volume groups on %s\n", current_config.disk);
+            on_error ();
+            return;
+        }
 
-                if (lvm_device == null) {
-                    warning ("unable to find 'data' volume group on %s\n", current_config.disk);
-                    on_error ();
-                    return;
-                }
+        unowned Distinst.LvmDevice lvm_device = disks.find_logical_volume ("data");
+
+        if (lvm_device == null) {
+            warning ("unable to find 'data' volume group on %s\n", current_config.disk);
```

Comment:
default `warning` behavior is already appending a \n
This sounds like a `critical`